src/schematools/contrib/django/factories.py

```diff
@@ -21,6 +23,7 @@
 )
 
 TypeAndSignature = Tuple[Type[models.Field], tuple, Dict[str, Any]]
+MODEL_CREATION_COUNTER = 1
```

Comment:
Not sure how the dynamic models cache works (per process, per thread, something else), but this will almost certainly require thread/process synchronization (locking)